hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/ClusterServiceImpl.java

```diff
@@ -233,7 +233,7 @@ void doRemoveAddress(Address deadAddress, boolean destroyConnection) {
             }
             Connection conn = node.connectionManager.getConnection(deadAddress);
             if (destroyConnection && conn != null) {
-                node.connectionManager.destroyConnection(conn);
+                conn.close(reason, null);
```

Comment:
Okay, you changed the internal calls from `TcpIpConnectionManager::destroyConnection()` to a new method `TcpIpConnectionManager::onClose()`, which handles the `closedCount.inc()`. I think this should be fine then.